source/common/network/connection_impl.cc

```diff
@@ -693,7 +738,7 @@ void ConnectionImpl::onReadReady() {
       onRead(new_buffer_size);
     }
     setDetectedCloseType(DetectedCloseType::RemoteReset);
-    closeSocket(ConnectionEvent::RemoteClose);
+    remoteCloseThroughFilterManager();
```

Comment:
We capture the reset on the read path, but ignore the reset on the write path, why?

Reply:
When I added the reset support before, and there is a flaky case that a Read is still needed before RST the connection #29616 and #28817, so in order to let the previous read finish we need to let it managed.